crates/languages/src/go/runnables.scm

```diff
@@ -1,9 +1,23 @@
+; Functions names start with `Test`
 (
-    (function_declaration name: (_) @run
-        (#match? @run "^Test.*"))
+    (function_declaration name: (_) @run @_name
+        (#match? @_name "^Test.+"))
 ) @go-test
 
+; `t.Run`
 (
-    (function_declaration name: (_) @run
-        (#eq? @run "main"))
-) @go-main
+    (call_expression function: (_) @run @_name
+        (#match? @_name "^t.Run.*"))
```

Comment:
Is it safe to assume that the variable we're calling `Run` on is always going to be called `t`? I get that it's a convention, but is there some precedent in Go community for breaking it sometimes?

Reply:
Ok, I think we can rework this pattern altogether to simplify it a bit:
With the following query:
```

; `t.Run`
(
    (call_expression 
      function: (
      	selector_expression
      	field: _ @run @_name
        (#eq? @_name "Run")
      )
      arguments: (
      	argument_list
        .
        (interpreted_string_literal ) @_subtest_name
       )
    )
    (#set! tag go-subtest)
)
```
In `build_context` the task_variables argument is gonna have the value of first argument (string literal) in `_subtest_name` key. You won't have to parse the line then. 
You still need to escape it (I guess?) and set it in the context being returned from build_context (captures with _ prefix are removed after a call to build_context so that end users can't reference it).

Reply:
Chipping in to say it's a fairly strong convention and good starting point. I've seen some codebases use test helpers that break the `t.Run` pattern but VSCode and GoLand don't match those subtests either.

Reply:
Ah, that's fair. With that other query (https://github.com/zed-industries/zed/pull/12003#discussion_r1613415317) we don't even need to match on variable name though, as we'll capture just the name of the function being called; so the whole dispute about whether we should match `t.` is kinda moot. ;p

Reply:
Would your example above match an unrelated `Run()` call like
```go
client.Run(ctx) // should not match

t.Run("client returns OK", func(t testing.T) {...}) // should match
```

Reply:
No, it would not. We also expect the first argument to be a string literal. We could perhaps constrain it further to also require a second parameter that's (I guess?) an anonymous function taking a single param of type `testing T`